### PR TITLE
Use EventArgs.Empty instead of new EventArgs()

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

**Comment by hughbe** [comment]
@zsd4yr I’m planning on adding tests in another PR, seems like the convention is that product and test changes come in different PRs

Essentially the PR removed unecessary allocations which is a win

**Comment by zsd4yr** [comment]
> @zsd4yr I’m planning on adding tests in another PR, seems like the convention is that product and test changes come in different PRs

We may be sending the wrong message here; perhaps we are making too many exceptions to the rule. This is not the desired paradigm. 

We prefer changes come _with_ tests. Please incorporate them into this PR

**Comment by karelz** [comment]
Bringing it to general discussion for a bit:
@hughbe I am a little bit surprised by your impression mentioned above - I wonder where it came from:

> I’m planning on adding tests in another PR, **seems like the convention is that product and test changes come in different PRs**

To my best knowledge, we do not have any convention for separating test changes from product changes - when there is a change, we expected related tests to be added ideally in the same change.
If there is large test gap, it is good idea to send PR for adding tests prior to sending PRs for product changes in the area (that's the only case when tests are ok to be separate from code changes IMO). Otherwise we might block the product changes until we have decent test coverage first.

If we somehow misled you to believe otherwise, I apologize. Maybe we didn't document / explain our expectations on our repos correctly - which is concerning to me.
If that's the case, please let me know - I'll be happy to follow up and find out way to clarify it for other contributors in reasonable way (our docs, guidance, etc.)
Thanks!

BTW: Thanks a lot for your GREAT contributions! We truly appreciate them - my question is just clarification on something which I suspect is just a misunderstanding - which is even more concerning that our seasoned contributors have, because it suggests fundamental lack of clarity on our side. Understanding your view point will help us to guide other contributors more & better in future. It is not reflection on any specific PR you provided ;)

**Comment by hughbe** [comment]
> If that's the case, please let me know - I'll be happy to follow up and find out way to clarify it for other contributors in reasonable way (our docs, guidance, etc.)

From the PR here (https://github.com/dotnet/winforms/pull/285#issuecomment-451559190) @zsd4yr said:
> 
> @hughbe after a discussion with the team today, we are asking you to separate this into three PRs, (1) which fixes those bugs you found, another (2) which cleans up these debugs, and a final (3) which contains the tests.

 > > Please let me know if you'd like any clarification about this decision.

Essentially this gave me the impression we should separate PRs into bug fixes/product changes, debug assertions and tests. I understand this seems weird but what I though the team wanted! Happy to change behaviour going forward

**Comment by zsd4yr** [comment]
Really sorry about the confusion here, @hughbe. I should have been more precise in that PR that our request was a deviation from the norms

**Comment by karelz** [comment]
Yeah, looks like an accidental confusion. Thanks for clarification @hughbe and thanks for being proactive at trying to follow repo-specific norms!
BTW: If you see interesting / weird differences between repos (CoreFX vs. WinForms/WPF), please let me know if future - those may be signs of things we want to normalize. Or at least check if those are intentional differences ;). Thanks!

**Comment by hughbe** [comment]
Will add tests when #476 and #365 are merged :))

**Comment by zsd4yr** [comment]
#476 and #365 are in.

**Comment by zsd4yr** [comment]
@hughbe please ping me here when this is ready for review again. Sorry about the delay getting back to it

**Comment by hughbe** [comment]
Rebased and added tests for those where I could - others are in message handers etc

File: src/System.Windows.Forms/src/System/Windows/Forms/ImageList.cs [hunk 1]
```diff
@@ -250,24 +250,24 @@ internal class Indexer {
...
+                    throw new ArgumentException(string.Format(SR.InvalidArgument, nameof(ImageSize), "Size.Empty"), nameof(value));
                 }
 
                 // ImageList appears to consume an exponential amount of memory
                 // based on image size x bpp.  Restrict this to a reasonable maximum
                 // to keep people's systems from crashing.
                 //
                 if (value.Width <= 0 || value.Width > maxImageWidth) {
-                    throw new ArgumentOutOfRangeException(nameof(ImageSize), string.Format(SR.InvalidBoundArgument, "ImageSize.Width", value.Width.ToString(CultureInfo.CurrentCulture), (1).ToString(CultureInfo.CurrentCulture), maxImageWidth.ToString()));
+                    throw new ArgumentOutOfRangeException(nameof(value), string.Format(SR.InvalidBoundArgument, "ImageSize.Width", value.Width.ToString(CultureInfo.CurrentCulture), (1).ToString(CultureInfo.CurrentCulture), maxImageWidth.ToString()));
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/ImageList.cs]: correction of param name

File: src/System.Windows.Forms/src/System/Windows/Forms/ImageList.cs [hunk 2]
```diff
@@ -250,24 +250,24 @@ internal class Indexer {
             }
             set {
                 if (value.IsEmpty) {
-                    throw new ArgumentException(string.Format(SR.InvalidArgument, "ImageSize", "Size.Empty"));
+                    throw new ArgumentException(string.Format(SR.InvalidArgument, nameof(ImageSize), "Size.Empty"), nameof(value));
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/ImageList.cs]: god bless...

I can't believe it was ever like this

**Pull request merged**